pkg/scheduler/core/generic_scheduler.go

```diff
@@ -650,17 +650,27 @@ func PrioritizeNodes(
 			results[i] = make(schedulerapi.HostPriorityList, len(nodes))
 		}
 	}
+	initializeHost := func(i int) {
+		for index, node := range nodes {
+			results[i][index].Host = node.Name
+		}
+	}
 	processNode := func(index int) {
 		nodeInfo := nodeNameToInfo[nodes[index].Name]
 		var err error
 		for i := range priorityConfigs {
+			// when error occurs, initialize the rest result[i] with default node name
+			if err != nil {
+				initializeHost(i)
+				continue
+			}
 			if priorityConfigs[i].Function != nil {
 				continue
 			}
 			results[i][index], err = priorityConfigs[i].Map(pod, meta, nodeInfo)
 			if err != nil {
 				appendError(err)
-				return
```

Comment:
please keep the return, as it's in the closure, so is as design.

Reply:
If `results[i]` returns with error, `return` will cause `results[i+1]` to be empty.

Reply:
I see.